backend/baseclass.pm

```diff
@@ -699,13 +699,26 @@ Returns the output on the serial device since the last call to set_serial_offset
 
 sub serial_text {
     my ($self) = @_;
+    return ($self->read_serial($self->{serial_offset}))[0];
+}
+
+=head2 read_serial
+
+Returns the output and the offset after reading on the serial device from position
+
+=cut
+
+sub read_serial {
+    my ($self, $position, $whence) = @_;
 
     open(my $SERIAL, "<", $self->{serialfile});
```

Comment:
I know this is not part of the changes, but i see no handling of error in case open fails :(

Reply:
there is, it's the `use autodie ':all';`